cosmetics-web/app/models/component.rb

```diff
@@ -112,6 +112,8 @@ def display_root_category
   end
 
   def nano_material_required?
+    return false if nano_material.nil?
```

Comment:
Not sure the naming of this method makes sense, as it seems it can be `true` if `nano_material` is already supplied. Maybe `nano_elements_required?` would be better. Not directly related to this change but since we're refactoring...

Also, could be more succinctly represented as:

```suggestion
    return nano_material && nano_material.nano_elements_required?
```

Reply:
I guess it’s really about validity, rather than whether they’re "required" or not?

Rename as `nano_material_valid?`? (And then we can refactor into an actual Rails validation at a later date).

Reply:
@slorek it's name isn't great, as this functionality was dealt with reactively. I've mentioned this needs to be ironed out by the ground but that's not happened yet.

I suggest there's a conversation about what a complete/incomplete notification looks like, then refactor these as appropriate.

At the moment nano elements are required if there's a nano material that does not have nano element information, so it seems to make sense in this context.

Reply:
> I guess it’s really about validity, rather than whether they’re "required" or not?
> 
> Rename as `nano_material_valid?`? (And then we can refactor into an actual Rails validation at a later date).

`*_valid` would be confusing. This isn't validation, so I don't think that's the appropriate course of action.